llvm/lib/Target/SPIRV/MCTargetDesc/SPIRVInstPrinter.cpp

```diff
@@ -314,6 +316,34 @@ void SPIRVInstPrinter::printOpDecorate(const MCInst *MI, raw_ostream &O) {
   }
 }
 
+void SPIRVInstPrinter::printUnknownType(const MCInst *MI, raw_ostream &O) {
+  const auto EnumOperand = MI->getOperand(1);
+  assert(EnumOperand.isImm() &&
+         "second operand of UNKNOWN_type must be opcode!");
+
+  const auto Enumerant = EnumOperand.getImm();
+  const auto NumOps = MI->getNumOperands();
+
+  // Encode the instruction enumerant and word count into the opcode
+  const auto OpCode = (0xFF & NumOps) << 16 | (0xFF & Enumerant);
+
+  // Print the opcode using the spirv-as unknown opcode syntax
+  O << "OpUnknown(" << Enumerant << ", " << NumOps << ") ";
```

Comment:
Is this already handled in spirv-as?

Reply:
It is now :)

https://github.com/KhronosGroup/SPIRV-Tools/pull/6024